run-queue

```diff
@@ -95,17 +95,22 @@ def consume_webhook_queue(channel, amqp):
 
 
 def consume_task_queue(channel, amqp, declare_public_result, declare_rhel_result, declare_stats_result):
-    queue = 'public'
-    if redhat_network():
-        # Try the rhel queue if the public queue is empty
-        if declare_public_result.method.message_count == 0:
-            queue = 'rhel'
-            # If both are non-empty, shuffle
-        elif declare_rhel_result.method.message_count > 0:
-            queue = ['public', 'rhel'][random.randrange(2)]
-
     if statistics_queue and declare_stats_result.method.message_count > 0:
-        queue = [queue, 'statistics'][random.randrange(2)]
+        # statistics queue is quick to process, always do that first
+        queue = 'statistics'
+    elif os.path.exists('/dev/kvm') or os.getenv('COCKPIT_TESTMAP_INJECT'):
```

Comment:
Should this be elif?

I understand it so far like this. Always add statistics queue if there are entries.

If /dev/kvm exists also appends to `queue` variable the other work. In order for this logic to take please we should replace this 'elif' with 'if' and also append to the `queue` array
Please let me know what I understand wrong.

Reply:
`queue` is not an array, it's a string. One can only call `basic_get()` on a particular queue. I don't want `queue = "statistics"` to be overwritten by e.g. `queue = "public"` in a task bot which has /dev/kvm and test queue entries, as I want "statistics" to have a higher priority than the tests.

Reply:
Note that this does not make much difference in practice -- we don't have infra which can run *both* statistics and tests. But we may have again one day (like the old CentOS CI cluster), and there this would be correct IMHO.